### DISCUSSION TITLE
Best way to set view theme when adding new view to workspace through contextmenu

### DISCUSSION THREAD

**Discussion by GREEB** [perspective-dev/perspective#1936] [Q&A] [answered]
Hello, 

I'm trying to set a theme of a viewer in a workspace when adding it by the "New Table>" drop down menu.

Whats the best way to do this? Thank you very much

**Comment by GREEB** [comment, marked as answer]
for now i fixed this with this. i'm sure there is a better way
```Javascript
    workspace.current.addEventListener(
      "workspace-new-view", async () => {

          if (workspace.current.children[0].attributes.getNamedItem('name') === null){           
            for (let i = 0; i < workspace.current.children.length; i++) {
              await workspace.current.children[i].restore({ theme: (theme === "light") ? "Material Light" : "Material Dark" })
            }
            await workspace.current.flush()
          }

      }
    );
```

update above version only works when workspace empty something like that would help if you add a new empty table to a workspace with views: 
```Javascript
 for (let index = 0; index < workspace.current.children.length; index++) {
          if (workspace.current.children[index].attributes.getNamedItem('name') === null) {
            const isTheme = (theme === "light") ? "Material Light" : "Material Dark"
            if (newView !== undefined && workspace.current !== null && workspace.current !== undefined) {
              if (workspace.current.children[index].attributes.getNamedItem('theme').value !== isTheme) {
                  await workspace.current.children[index].restore({ theme: (theme === "light") ? "Material Light" : "Material Dark" })
              }
            }
          }
        }
```

**Reply by texodus** [reply]
This is an intended use-case for the `"workspace-new-view"` Event!  So perfectly reasonably implementation for the API as it stands today.  I do intend to add complete theme support to `<perspective-workspace>`, opened an issue to track this #1937.

^ This would only be applicable for changing themes _at runtime_ though, or choosing a different theme for each new `perspetive-viewer` than the container's theme;  the existing `<perspective-workspace>` themes shouldn't need special support to properly apply their style to new `<perspective-viewer>`s added via the context menu.  For example, the [`movies` example projects using the "Material Dark" theme](https://github.com/finos/perspective/blob/master/examples/blocks/src/movies/index.html#L5).

**Reply by GREEB** [reply]
Great thanks a lot, its just for light/dark website themes, i use the material themes and switch themes in runtime no problem the only problem i had was when adding a new table from context-menu. I guess because of my setup i load both and when adding a new table it defaults to the first theme, and its the light one.

It still flashes for a frame or two but that could be hidden easily.

Little demo but you don't even see the flashing because of frame rate
![Peek 2022-08-04 18-40](https://user-images.githubusercontent.com/1221769/182904893-1480e01a-9deb-4755-90a4-45063cc41d63.gif)
![Peek 2022-08-04 19-04](https://user-images.githubusercontent.com/1221769/182909442-663a5802-d1f2-4b90-a523-615ac74c8537.gif)


btw. great work on perspective love it!
